CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.20)
+
+# Must be done first
+if (APPLE)
+    # Force clang compiler on macOS
+    set(CMAKE_CXX_COMPILER "/usr/bin/clang++")
+    set(CMAKE_C_COMPILER "/usr/bin/clang")
+endif ()
+
+# Options
+option(WITH_TLS "Build valkey-server with TLS support" OFF)
+option(BUILD_TESTS "Build valkey-unit-tests" OFF)
+option(WITH_RDMA "Build valkey-rdma module" OFF)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+project("valkey")
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS OFF)
```

Comment:
This is in theory more restrictive than what we have today: `-std=gnu11` is C11 plus gnu extensions I believe. I am surprised that this doesn't break the build? can you double check?

Reply:
The Makefile has this:

```make
 40 # Use -Wno-c11-extensions on clang, either where explicitly used or on
 41 # platforms we can assume it's being used.
 42 ifeq (clang,$(CLANG))
 43   STD+=-Wno-c11-extensions
 44 else
 45 ifneq (,$(findstring FreeBSD,$(uname_S)))
 46   STD+=-Wno-c11-extensions
 47 endif
 48 endif
```

while `-std=c99` and `-std=gnu11` is set by the detection of `stdatomic.h` as seen here:

```make
 52 # Detect if the compiler supports C11 _Atomic.
 53 # NUMBER_SIGN_CHAR is a workaround to support both GNU Make 4.3 and older versions.
 54 NUMBER_SIGN_CHAR := \#
 55 C11_ATOMIC := $(shell sh -c 'echo "$(NUMBER_SIGN_CHAR)include <stdatomic.h>" > foo.c; \
 56     $(CC) -std=gnu11 -c foo.c -o foo.o > /dev/null 2>&1; \
 57     if [ -f foo.o ]; then echo "yes"; rm foo.o; fi; rm foo.c')
 58 ifeq ($(C11_ATOMIC),yes)
 59     STD+=-std=gnu11
 60 else
 61     STD+=-std=c99
 62 endif
```

Reply:
The above check is implemented in `ValkeySetup.cmake` here:
https://github.com/eifrah-aws/valkey/blob/cmake/cmake/Modules/ValkeySetup.cmake#L233